File: data-structure-typescript/src/list/common/list.interface.ts
export interface List<T> {
  add (element: T): void;

  add (
    index: number,
    element: T,
  ): void;

  remove (index: number): T;

  get (index: number): T;

  set (
    index: number,
    element: T,
  ): T;

  indexOf (element: T): number;

  size (): number;

  isEmpty (): boolean;

  clear (): void;

  toArray (): T[];
}

File: data-structure-java/src/main/java/com/wisoft/io/datastructure/list/common/ListInterface.java
package com.wisoft.io.datastructure.list.common;

public interface ListInterface<T> {
    void add(T element);

    void add(int index, T element);

    T remove(int index);

    T get(int index);

    T set(int index, T element);

    int indexOf(T element);

    int size();

    boolean isEmpty();

    void clear();

    Object[] toArray();
}
